gm4_vertical_rails/data/gm4_vertical_rails/functions/rail/apply_direction.mcfunction

```diff
@@ -0,0 +1,11 @@
+# Checks if Vertical Rails go up or downward, prioritising up
+# @s = minecart, touching a Vertical Rail (ladder), with gm4_vert_rails_momentum tag
+# at @s
+# run from process
+
+# if going up a Vertical Rail
+execute as @s[tag=gm4_vert_rail_up] at @s run function gm4_vertical_rails:rail/upward/momentum
+
+# if going down a Vertical Rail, or at bottom of Rail
+execute as @s[tag=gm4_vert_rail_down] at @s if block ~ ~-1 ~ ladder unless block ~ ~1 ~ ladder run function gm4_vertical_rails:rail/downward/momentum
+execute as @s[tag=gm4_vert_rail_down] at @s unless block ~ ~-1 ~ ladder run function gm4_vertical_rails:rail/downward/detach_ladder
```

Comment:
The amount of reference frame changes in this function can be reduced to 0. See my comments attached to `check_direction.mcfunction`.
The momentum changes shouldn't alter the carts position until the next tick, so re-calling `at @s` shouldn't be necessary.